client/landing/stepper/declarative-flow/internals/steps-repository/playground/lib/blueprint.ts

```diff
@@ -179,6 +175,64 @@ async function getBlueprintFromUrl( recommendedPhpVersion: string ): Blueprint {
 	};
 }
 
+async function getBlueprintFromUrl( recommendedPhpVersion: string ): Blueprint | BlueprintBundle {
+	const url = new URL( window.location.href );
+	const predefinedBlueprintName = getBlueprintName( url.searchParams.get( 'blueprint' ) );
+
+	// If a predefined blueprint is specified and exists, use it
+	if ( predefinedBlueprintName ) {
+		const blueprint = PREDEFINED_BLUEPRINTS[ predefinedBlueprintName ];
+		return {
+			...blueprint,
+			preferredVersions: {
+				...blueprint.preferredVersions,
+				php: getPHPVersion( recommendedPhpVersion ),
+			},
+		};
+	}
+
+	let blueprint: Blueprint | BlueprintBundle = {};
+	if ( url.searchParams.has( 'blueprint-url' ) ) {
+		blueprint = await resolveRemoteBlueprint( url.searchParams.get( 'blueprint-url' )! );
+	}
+
+	if ( isBlueprintBundle( blueprint ) ) {
+		const blueprintDeclaration = await getBlueprintDeclaration( blueprint );
+		const mergedBlueprint = deeplyMergeBlueprintWithDefaults(
+			blueprintDeclaration,
+			recommendedPhpVersion
+		);
+		/**
+		 * Blueprint Bundles don't support writing to the filesystem, so we need to modify the read method
+		 * to return a modified blueprint.json file that includes the merged blueprint.
+		 *
+		 * This is a workaround and should be removed once Playground starts allowing
+		 * Blueprint Bundles to be modify the blueprint.json file.
+		 */
+		const originalRead = blueprint.read.bind( blueprint );
+		blueprint.read = async ( path: string ) => {
+			const normalizedPath = path.startsWith( '/' ) ? path.slice( 1 ) : path;
+			if ( normalizedPath === 'blueprint.json' ) {
+				const jsonString = JSON.stringify( mergedBlueprint, null, 2 );
+				const bytes = new TextEncoder().encode( jsonString );
+				const stream = new ReadableStream< Uint8Array >( {
+					start( controller ) {
+						controller.enqueue( bytes );
+						controller.close();
+					},
+				} );
+				return new StreamedFile( stream, 'blueprint.json', {
+					type: 'application/json',
+					filesize: bytes.length,
+				} );
+			}
+			return originalRead( path );
+		};
+		return blueprint;
```

Comment:
@adamziel is there a recommended way to modify a bundled blueprint.json file?

If not, we could add a write method or similar to [the `ZipFilesystem` class](https://github.com/WordPress/wordpress-playground/blob/5a2e706b345f579878388bdd7531e756af697942/packages/playground/storage/src/lib/filesystems.ts#L55)?